src/main/java/de/tum/in/www1/artemis/security/jwt/JWTFilter.java

```diff
@@ -32,25 +32,54 @@ public JWTFilter(TokenProvider tokenProvider) {
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
         HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
         Cookie jwtCookie = WebUtils.getCookie(httpServletRequest, JWT_COOKIE_NAME);
-        if (isJwtCookieValid(this.tokenProvider, jwtCookie)) {
-            Authentication authentication = this.tokenProvider.getAuthentication(jwtCookie.getValue());
+
+        // check if valid JWT token is in the cookie or in the Authorization header
+        // then proceed to do authentication with this token
+        String jwtToken;
+        if (isJwtValid(this.tokenProvider, jwtToken = getJwtFromCookie(jwtCookie))
```

Comment:
not doing it that way would lead to a lot of encapsuled if else statements. i think this code style with the according makes it easier readable than the if else statements.

Reply:
we decided to make a feature branch for all vscode plugin related issues and make one bigger PR then